tests/cephfs/subvolume_authorize.py

```diff
@@ -0,0 +1,426 @@
+            kernel_mount_dir = "/mnt/" + "".join(
+                secrets.choice(string.ascii_uppercase + string.digits) for i in range(5)
+            )
+            fuse_mount_dir = "/mnt/" + "".join(
+                secrets.choice(string.ascii_lowercase + string.digits) for i in range(5)
+            )
+            mount_points.extend([kernel_mount_dir, fuse_mount_dir])
+            log.info("Testing client with read-write permission on subvolume")
+            fs_util.subvolume_authorize(
+                client[0],
+                fs_name,
+                subvolume_name,
+                client_name,
+                extra_params=f" {subvolgroup_name} --access_level=rw",
+            )
+            out, rc = client[0].exec_command(
+                sudo=True,
+                cmd=f"ceph fs subvolume authorized_list {fs_name} {subvolume_name} {subvolgroup_name}",
+            )
+            output = out.read().decode().strip()
+            if client_name in output:
+                log.info("Client creation successful")
+            else:
+                log.error("Client creation failed")
+                return 1
+            log.info("Getting the path of Cephfs subvolume")
+            subvol_path, rc = client[0].exec_command(
+                sudo=True,
+                cmd=f"ceph fs subvolume getpath {fs_name} {subvolume_name} {subvolgroup_name}",
+            )
+            subvol_path = subvol_path.read().decode().strip()
+            log.info(f"Testing kernel & fuse mount for {client_name}")
+            fs_util.kernel_mount(
+                client,
+                kernel_mount_dir,
+                mon_node_ip,
+                new_client_hostname=client_name,
+                sub_dir=f"{subvol_path}",
+            )
+            fs_util.fuse_mount(
+                client,
+                fuse_mount_dir,
+                new_client_hostname=client_name,
+                extra_params=f" -r {subvol_path}",
+            )
+            log.info(f"Testing read-write operation on {client_name}")
+            commands = [
+                f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation create --threads 10 --file-size 4 "
+                f"--files 1000 --files-per-dir 10  --top {kernel_mount_dir}",
+                f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation read --threads 10 --file-size 4 "
+                f"--files 1000 --files-per-dir 10 --top {kernel_mount_dir}",
+                f"dd if=/dev/zero of={fuse_mount_dir}/file bs=10M count=10",
+                f"dd if={fuse_mount_dir}/file of={fuse_mount_dir}/file2 bs=10M count=10",
+            ]
+            for command in commands:
+                _, err = client[0].exec_command(
+                    sudo=True, cmd=command, long_running=True
+                )
+                if err:
+                    log.error(f"Permissions set for {client_name} is not working")
+                    return 1
```

Comment:
I believe we have used the same steps in another test module. Can we implement a method and reuse it here?

Reply:
Sure